### PR TITLE
Take LoRa coding rate from PHY and relax DR format

### PR DESCRIPTION
**Pull request by johanstokking**
**Summary:**

References https://github.com/TheThingsIndustries/lorawan-stack/issues/1649

**Changes:**

- Use LoRa coding rate from PHY
- Remove obsolete TODO, use `phy` for band instance

### CODE REVIEW

File: pkg/ttnpb/udp/data_rate.go [hunk 1]
```diff
@@ -67,8 +67,8 @@ func (d *DataRate) UnmarshalJSON(data []byte) error {
 
 var (
 	errDataRate = errors.DefineInvalidArgument("data_rate", "invalid data rate")
-	sfRegexp    = regexp.MustCompile("^SF(6|7|8|9|10|11|12)")
-	drRegexp    = regexp.MustCompile("BW(125|250|500)$")
+	sfRegexp    = regexp.MustCompile("^SF(5|6|7|8|9|10|11|12)")
```
Comment (htdvisser) [inline pkg/ttnpb/udp/data_rate.go]: Shouldn't we just make this a `\d+`?

File: pkg/ttnpb/udp/data_rate.go [hunk 2]
```diff
@@ -67,8 +67,8 @@ func (d *DataRate) UnmarshalJSON(data []byte) error {
...
+	drRegexp    = regexp.MustCompile("BW(\\d{2,3})$")
```
Comment (htdvisser) [inline pkg/ttnpb/udp/data_rate.go]: Shouldn't we just make this a `\d+`? Perhaps even add a `(?:\.\d+)?`. The gateway module also supports `62.5`, `31.25`, `15.625`, `7.8125`, so I can imagine those may be used in the future too.

File: pkg/band/band.go [hunk 3]
```diff
@@ -173,6 +173,9 @@ type Band struct {
 	// DefaultMaxEIRP in dBm
 	DefaultMaxEIRP float32
 
+	// LoRaCodingRate is the coding rate used for LoRa modulation.
+	LoRaCodingRate string
```
Comment (htdvisser) [inline pkg/band/band.go]: While we're at it, we should perhaps also add the sync word

Comment (johanstokking) [reply pkg/band/band.go]: Let's do that when we need it and there's a way to control it.

- UDP packet forwarder flavors do not support setting the sync word not a `public` flag
- Basic Station does not support setting the sync word either, but does allow for `lorawan_public`, probably changing the sync word
- The Things Gateway (Kickstarter) supports setting the sync word but 

When things converge, we'll see what we do

**Pull request merged**